src/lua/init.lua

```diff
@@ -349,13 +349,31 @@ local function run_preload()
     end
 end
 
-return {
-    uptime = uptime;
-    pid = pid;
-    compat = compat;
+-- Extract all fields from a table except ones that start from
+-- the underscore.
+--
+-- Useful for __serialize.
+local function hide_internal_fields(t)
```

Comment:
Minor: I only wonder, why this function is not simply called `serialize`, but I'm not going to argue regarding the naming here. Anyway, my two cents here:
1. The function doesn't work in-place, but IMHO "hide" implies this.
2. You **get** the new table containing only the **public fields**.

It looks like `get_public_fields` fits more here. Feel free to ignore.

Reply:
> I only wonder, why this function is not simply called `serialize`, but I'm not going to argue regarding the naming here.

I've named the function according to what is does, not how it is used. It seems meaningful for me.

> 1. The function doesn't work in-place, but IMHO "hide" implies this.

Maybe. I'll think here a bit more.

> 2. You **get** the new table containing only the **public fields**.
>
> It looks like `get_public_fields` fits more here.

I don't like `get_` functions -- it looks like one failed to find an appropriate name to describe what it does. I wouldn't say that all the remaining fields are definitely public. I know that `_<...>` ones are internal and don't know anything about others.

Reply:
Renamed to `filter_out_private_fields()` (it also follows a terminology from #7594 and such fields are actually called 'private' in other languages).